docs/content/modeling/abac-conditions.mdx

```diff
@@ -0,0 +1,226 @@
+  authorizationModel={{
+    "schema_version":"1.1",
+    "type_definitions": [
+      {
+        "type":"user"
+      },
+      {
+        "type":"document",
+        "relations": {
+          "viewer": {
+            "this": {}
+          }
+        },
+        "metadata": {
+          "relations": {
+            "viewer": {
+              "directly_related_user_types": [
+                {
+                    "type":"user",
+                    "condition":"temporal_grant"
+                }
+              ]
+            }
+          }
+        }
+      }
+    ],
+    "conditions": {
+      "temporal_grant": {
+        "name":"temporal_grant",
+        "expression":"current_time <= grant_time + grant_duration",
+        "parameters": {
+          "current_time": {
+              "type_name":"TYPE_NAME_TIMESTAMP"
+          },
+          "grant_duration": {
+              "type_name":"TYPE_NAME_DURATION"
+          },
+          "grant_time": {
+              "type_name":"TYPE_NAME_TIMESTAMP"
+          }
+        }
+      }
+    }
+  }}
+  allowedLanguages={[
+    SupportedLanguage.JS_SDK,
+    SupportedLanguage.GO_SDK,
+    SupportedLanguage.DOTNET_SDK,
+    SupportedLanguage.PYTHON_SDK,
+    SupportedLanguage.CLI,
+    SupportedLanguage.CURL,
+  ]}
+/>
+
+## Writing Conditional Relationship Tuples
+Using the model above, when we [Write](https://openfga.dev/api/service#/Relationship%20Tuples/Write) relationship tuples to the OpenFGA store, then any `document#viewer` relationship with `user` objects must be accompanied by the condition `temporal_grant` because the type restriction requires it.
+
+For example, we can give `user:anne` viewer access to `document:1` for 10 minutes by writing the following relationship tuple:
+
+<WriteRequestViewer
+  relationshipTuples={[
+    {
+      user: 'user:anne',
+      relation: 'viewer',
+      object: 'document:1',
+      condition: {
+        name: 'temporal_grant',
+        context: {
+          grant_time: '2023-01-01T00:00:00Z',
+          grant_duration: '10m'
+        }
+      }
+    }
+  ]}
+  skipSetup={true}
+  allowedLanguages={[
+    SupportedLanguage.JS_SDK,
+    SupportedLanguage.GO_SDK,
+    SupportedLanguage.DOTNET_SDK,
+    SupportedLanguage.PYTHON_SDK,
+    SupportedLanguage.CLI,
+    SupportedLanguage.CURL,
+  ]}
+/>
+
+## Queries with Condition Context
+Now that we have written a [Conditional Relationship Tuple](../concepts#what-is-a-conditional-relationship-tuple), we can query OpenFGA using the [Check API](https://openfga.dev/api/service#/Relationship%20Queries/Check) to see if `user:anne` has viewer access to `document:1` under certain conditions/context. That is, `user:anne` should only have access is the current timestamp is less than or equal to the grant timestamp (e.g. the time which the tuple was written) plus the duration of the grant (10 minutes). If the current timestamp is less than or equal, then you'll get a permit decision. For example,
+
+<CheckRequestViewer
```

Comment:
todo: CheckRequestViewer needs to be wired up to support `context`. It does not currently have support at this time.